### PR TITLE
spawn electrs in a seperate process group so SIGHUPs don't stop it

### PR DESCRIPTION
**Pull request by gandrewstone**

### CODE REVIEW

File: src/utilprocess.cpp [hunk 1]
```diff
@@ -168,7 +168,14 @@ void SubProcess::Run()
     }
     cmdargs[cmdargs.size() - 1] = nullptr;
     pid_t child_pid;
-    THROW_IF_RETERROR(posix_spawnp(&child_pid, path.c_str(), &action, nullptr, &cmdargs[0], nullptr));
+    posix_spawnattr_t attr;
+    posix_spawnattr_init(&attr);
```
Comment (dagurval) [inline src/utilprocess.cpp]: Wrap in THROW_IF_RETERROR

File: src/utilprocess.cpp [hunk 2]
```diff
@@ -168,7 +168,14 @@ void SubProcess::Run()
...
+    posix_spawnattr_setflags(&attr, POSIX_SPAWN_SETPGROUP);
```
Comment (dagurval) [inline src/utilprocess.cpp]: This as well

File: src/utilprocess.cpp [hunk 3]
```diff
@@ -168,7 +168,14 @@ void SubProcess::Run()
...
+    posix_spawnattr_setpgroup(&attr, 0);
```
Comment (dagurval) [inline src/utilprocess.cpp]: And this

File: src/utilprocess.cpp [hunk 4]
```diff
@@ -168,7 +168,14 @@ void SubProcess::Run()
...
+
+    THROW_IF_RETERROR(posix_spawnp(&child_pid, path.c_str(), &action, &attr, &cmdargs[0], nullptr));
+
+    posix_spawnattr_destroy(&attr);
```
Comment (dagurval) [inline src/utilprocess.cpp]: This should be in the destructor of RunCleanupRAII

**Review by dagurval — COMMENTED** [review]
LGTM after nits

**Comment by gandrewstone** [comment]
@dagurval It didn't make sense to put pthread_spawnattr_destroy in the existing RAII because the lifecycle is different.  In particular, it can be freed after the spawn, rather than being held for the full duration of the program's run.  So I created a separate RAII class.  

I added THROW_IF_RETERROR to everything except the _destroy call.  The problem with doing it around _destroy is if the code calls the RAII destructor because a different exception occurred then throwing on _destroy failure would hide the real problem.  I doubt that _destroy can fail, but if it does, we leak a few bytes in the worst case (and there's nothing a handler could do about this since attr would be out of scope when the exception handler executes).  So I think its actually better to go on rather than abort().

**Review by dagurval — APPROVED** [review]
👍

**Pull request merged**